hermes/probe/create/create_file_test.go

```diff
@@ -0,0 +1,140 @@
+			}
+		}
+	}
+}
+
+func TestChecksum(t *testing.T) {
+	f := RandomHermesFile{11, 100}
+	fTwo := RandomHermesFile{13, 1000}
+	fCopy := RandomHermesFile{11, 100}
+	checksum, err := f.CheckSum()
+	if err != nil {
+		t.Error(err)
+	}
+	checksumTwo, err := fTwo.CheckSum()
+	if err != nil {
+		t.Error(err)
+	}
+
+	checksumCopy, err := fCopy.CheckSum()
+	if err != nil {
+		t.Error(err)
+	}
+
+	if fmt.Sprintf("%x", checksum) != fmt.Sprintf("%x", checksumCopy) { //comparing checksums converted to strings as a slice can only be compared to nil
+		t.Errorf("Checksum returned different values for the same RandomHermesFiles {%v, %v}", f.ID, f.Size)
+	}
+
+	if fmt.Sprintf("%x", checksum) == fmt.Sprintf("%x", checksumTwo) {
+		t.Errorf("Checksum returned the same value for two different RandomHermesFiles {%v, %v} and {%v, %v}", f.ID, f.Size, fTwo.ID, fTwo.Size)
+	}
+
+}
+
+func TestCreateFile(t *testing.T) {
+	ctx := context.Background()
+	bucketName := "test_bucket_probe0"
+	client := mock.NewFakeClient()
+	fbh := client.Bucket(bucketName)                         // fakeBucketHandle
+	if err := fbh.Create(ctx, bucketName, nil); err != nil { // creates the bucket with name "test_bucket_probe0"
+		t.Error(err)
+	}
+	fileID := int32(6)
+	fileSize := 50
+	target := &probe.Target{
+		&probepb.Target{
+			Name:                   "hermes",
+			TargetSystem:           probepb.Target_GOOGLE_CLOUD_STORAGE,
+			TotalSpaceAllocatedMib: int64(1000),
+			BucketName:             "test_bucket_probe0",
+		},
+		&journalpb.StateJournal{
+			Filenames: make(map[int32]string),
+		},
+		&metrics.Metrics{},
+	}
+	hp := &probepb.HermesProbeDef{
+		ProbeName: proto.String("createfile_test"),
+		Targets: []*probepb.Target{
+			&probepb.Target{
+				Name:                   "hermes",
+				TargetSystem:           probepb.Target_GOOGLE_CLOUD_STORAGE,
+				TotalSpaceAllocatedMib: int64(100),
+				BucketName:             "test_bucket_probe0",
+			},
+		},
+		TargetSystem: probepb.HermesProbeDef_GCS.Enum(),
+		IntervalSec:  proto.Int32(3600),
+		TimeoutSec:   proto.Int32(60),
+		ProbeLatencyDistribution: &metricpb.Dist{
+			Buckets: &metricpb.Dist_ExplicitBuckets{
+				ExplicitBuckets: "0.1,0.2,0.4,0.6,0.8,1.6,3.2,6.4,12.8,1",
+			},
+		},
+		ApiCallLatencyDistribution: &metricpb.Dist{
+			Buckets: &metricpb.Dist_ExplicitBuckets{
+				ExplicitBuckets: "0.1,0.2,0.4,0.6,0.8,1.6,3.2,6.4,12.8,1",
+			},
+		},
+	}
+	probeTarget := &probepb.Target{
+		Name:                   "hermes",
+		TargetSystem:           probepb.Target_GOOGLE_CLOUD_STORAGE,
+		TotalSpaceAllocatedMib: int64(100),
+		BucketName:             "test_bucket_probe0",
+	}
+
+	err := fmt.Errorf("") //err cannot be defined in the if statement below using ':=' as then the set up fails with the error:  expected identifier on left side of :=
+	if target.LatencyMetrics, err = metrics.NewMetrics(hp, probeTarget); err != nil {
```

Comment:
If I understand it, target.LatencyMetrics is not used. If that's the case, then use the `_` placeholder to throw away that return value. Then you can skip the `var err error` part and use `:=` instead: `if _, err := ...; err != nil {`.

Reply:
Maybe I misunderstood it but I think that here metrics are initialised in the LatencyMetrics field in target (which is a Target struct - defined in package probe). Is this right @evanSpendlove ? That's why I couldn't use ':=@ and had to define error beforehand

Reply:
target.LatencyMetrics is normally initialised inside the probe.go file. However, this is a test environment that doesn't use probe.go.

target.LatencyMetrics is used in create_file.go so it needs to be initialised before target is passed to CreateFile().

Reply:
E.g. line 128 of create_file.go

Reply:
I see! This is a set-up operation.
In this case use t.Fatalf() on failure.

Reply:
Changed it to t.Fatalf()